lambdas/answer/src/test/java/uk/gov/di/ipv/cri/kbv/api/handler/QuestionAnswerHandlerTest.java

```diff
@@ -82,12 +86,23 @@ class QuestionAnswerHandlerTest {
     @Mock private ConfigurationService mockConfigurationService;
     @Mock private AuditService mockAuditService;
     @Mock private KBVGateway mockKBVGateway;
+    @Mock private ServiceFactory mockServiceFactory;
+    @Mock private KBVClientFactory mockKBVClientFactory;
     @Captor private ArgumentCaptor<Map<String, Object>> auditEventExtensionsArgCaptor;
 
     @BeforeEach
     void setUp() {
+        lenient().when(mockServiceFactory.getKbvGateway()).thenReturn(mockKBVGateway);
+        lenient()
+                .when(mockServiceFactory.getKbvClientFactory(MOCK_CLIENT_ID))
+                .thenReturn(mockKBVClientFactory);
+        lenient()
+                .when(mockKBVClientFactory.createClient(any()))
+                .thenReturn(mock(IdentityIQWebServiceSoap.class));
+
```

Comment:
I suppose if you make the soap cache thing injectable you won't need these. I try to avoid lenient stubbing not to say there isn't a case for it. But i think this is symptom of not injecting the soap cache thing

Reply:
I am using `lenient` because not all the tests need the stubbing but most of them do. It's either we have the duplication or use lenient. Unless you can see a better way, I don't particularly have a problem with this approach.

Reply:
Done - thanks Ola